nvdaHelper/remote/excel.cpp

```diff
@@ -0,0 +1,438 @@
+							if(FAILED(res)) {
+								LOG_DEBUGWARNING(L"columns.item "<<colCount<<L" failed with code "<<res);
+							}
+							if(pDispatchLastColumn) {
+								res=_com_dispatch_raw_propget(pDispatchLastColumn,XLDISPID_RANGE_NEXT,VT_DISPATCH,&pDispatchNextCell);
+								if(FAILED(res)) {
+									LOG_DEBUGWARNING(L"range.next failed with code "<<res);
+								}
+							}
+						}
+					}
+				}
+				if(rangeWidth==0) { // could not get width from a merge area
+					res=_com_dispatch_raw_propget(pDispatchRange,XLDISPID_RANGE_WIDTH,VT_I4,&rangeWidth);
+					if(FAILED(res)) {
+						LOG_DEBUGWARNING(L"range.width failed with code "<<res);
+					}
+				}
+				if(textWidth>rangeWidth) {
+					if(!pDispatchNextCell) {
+						res=_com_dispatch_raw_propget(pDispatchRange,XLDISPID_RANGE_NEXT,VT_DISPATCH,&pDispatchNextCell);
+						if(FAILED(res)) {
+							LOG_DEBUGWARNING(L"range.next failed with code "<<res);
+						}
+					}
+					if(pDispatchNextCell) {
+						CComBSTR text;
+						res=_com_dispatch_raw_propget(pDispatchNextCell,XLDISPID_RANGE_TEXT,VT_BSTR,&text);
+						if(FAILED(res)) {
+							LOG_DEBUGWARNING(L"range.text failed with code "<<res);
+						}
+						if(text&&text.Length()>0) {
+							states|=NVSTATE_CROPPED;
+						}
+					}
+					if(!(states&NVSTATE_CROPPED)) {
+						states|=NVSTATE_OVERFLOWING;
+					}
+				}
+			}
+		}
+	}
+	return states;
+}
+
+HRESULT getCellInfo(HWND hwnd, IDispatch* pDispatchRange, long cellInfoFlags, EXCEL_CELLINFO* cellInfo) {
+	HRESULT res=S_OK;
+	if(cellInfoFlags&NVCELLINFOFLAG_TEXT) {
+		res=_com_dispatch_raw_propget(pDispatchRange,XLDISPID_RANGE_TEXT,VT_BSTR,&cellInfo->text);
+		if(FAILED(res)) {
+			LOG_DEBUGWARNING(L"range.text failed with code "<<res);
+		}
+	}
+	if(cellInfoFlags&NVCELLINFOFLAG_FORMULA) {
+		res=_com_dispatch_raw_propget(pDispatchRange,XLDISPID_RANGE_FORMULA,VT_BSTR,&cellInfo->formula);
+		if(FAILED(res)) {
+			LOG_DEBUGWARNING(L"range.formula failed with code "<<res);
+		}
+	}
+	if(cellInfoFlags&NVCELLINFOFLAG_COMMENTS) {
+		CComPtr<IDispatch> pDispatchComment=nullptr;
+		res=_com_dispatch_raw_propget(pDispatchRange,XLDISPID_RANGE_COMMENT,VT_DISPATCH,&pDispatchComment);
+		if(FAILED(res)) {
+			LOG_DEBUGWARNING(L"range.comment failed with code "<<res);
+		}
+		if(pDispatchComment) {
+			res=_com_dispatch_raw_method(pDispatchComment,XLDISPID_COMMENT_TEXT,DISPATCH_METHOD,VT_BSTR,&cellInfo->comments,nullptr);
+			if(FAILED(res)) {
+				LOG_DEBUGWARNING(L"comment.text failed with code "<<res);
+			}
+		}
+	}
+	if(cellInfoFlags&NVCELLINFOFLAG_INPUTMESSAGE) {
+		CComPtr<IDispatch> pDispatchValidation=nullptr;
+		res=_com_dispatch_raw_propget(pDispatchRange,XLDISPID_RANGE_VALIDATION,VT_DISPATCH,&pDispatchValidation);
+		if(FAILED(res)) {
+			LOG_DEBUGWARNING(L"range.validation failed with code "<<res);
+		}
+		if(pDispatchValidation) {
+			_com_dispatch_raw_propget(pDispatchValidation,XLDISPID_VALIDATION_INPUTTITLE,VT_BSTR,&cellInfo->inputTitle);
```

Comment:
The results of this com call and the calls below are not yet logged on failure.

Reply:
All COM calls now check for errors. Plus I removed a duplicate call to range.address which I'd accidentally left in after a previous refactor.